app/helpers/interfaces_helper.rb

```diff
@@ -4,4 +4,8 @@ def interfaces_types
       [_(nic_type.humanized_name), nic_type]
     end
   end
+
+  def virtual_host?(nic)
+    %w(Ovirt Libvirt Vmware).map { |cm| nic&.host&.compute_resource&.type&.include?(cm) }.any?
```

Comment:
What if it's GCE/OpenStack provider?

Shouldn't we check all the providers (not only these 3) we support?

Reply:
+1, we tend to add `support_$feature` to ComputeResource parent and override that to true in children, e.g. https://github.com/theforeman/foreman/blob/develop/app/models/compute_resource.rb#L249-L252 or https://github.com/theforeman/foreman/blob/develop/app/models/compute_resource.rb#L393-L395

so something like `support_mac_assignment` would be good for checking, if the field should or should not be enabled

Reply:
I believe we already have this.

The compute resources have a provided method. Just ask the cr if it provides the mac. Note this will not work for virtual interfaces.